File: 283-move-zeroes/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        position = 0
        for i in range(0,len(nums)):
            if nums[i]!=0:
                nums[position] = nums[i]
                position += 1
            
        for i in range(position,len(nums)):
            nums[i] = 0
        """
        Do not return anything, modify nums in-place instead.
        """
        

File: 283-move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        
        int left =  0;
        for(int i = 0;i< nums.length;i++){
            
            while( i < nums.length && nums[i] == 0 ){
                i++;
            }
            if(i < nums.length)nums[left++] = nums[i];

        }
         for(int i = left ; i <nums.length;i++){
            nums[i] = 0;
         }
    }
}